modules/drivers/clickhouse/src/metabase/driver/clickhouse.clj

```diff
@@ -241,7 +263,17 @@
    {:write? true}
    (fn [^java.sql.Connection conn]
      (with-open [stmt (.createStatement conn)]
-       (.execute stmt (create-table!-sql driver table-name column-definitions :primary-key primary-key))))))
+       (let [sql (create-table!-sql driver table-name column-definitions :primary-key primary-key)]
+         (.execute stmt sql))))))
+
+(defmethod driver/rename-tables!* :clickhouse
+  [_driver db-id sorted-rename-map]
+  (jdbc/with-db-transaction [conn (sql-jdbc.conn/db->pooled-connection-spec db-id)]
+    (with-open [stmt (.createStatement ^java.sql.Connection (:connection conn))]
+      (doseq [[old-table-name new-table-name] sorted-rename-map]
```

Comment:
Adda TODO here to use locks or transactions or whatever if we can. Can link to that follow up issue.